src/thriveopedia/pages/wiki/OrganelleInfoBox.cs

```diff
@@ -0,0 +1,235 @@
+
+    [Export]
+    public NodePath ProcessesLabelPath = null!;
+
+    [Export]
+    public NodePath EnzymesLabelPath = null!;
+
+    [Export]
+    public NodePath MassLabelPath = null!;
+
+    [Export]
+    public NodePath SizeLabelPath = null!;
+
+    [Export]
+    public NodePath OsmoregulationCostLabelPath = null!;
+
+    [Export]
+    public NodePath StorageLabelPath = null!;
+
+    [Export]
+    public NodePath UniqueIconPath = null!;
+
+    [Export]
+    public NodePath UpgradesLabelPath = null!;
+
+    [Export]
+    public NodePath InternalNameLabelPath = null!;
+
+    private OrganelleDefinition? organelle;
+
+#pragma warning disable CA2213
+    private Label nameLabel = null!;
+    private TextureRect icon = null!;
+    private TextureRect? model;
+    private Label costLabel = null!;
+    private TextureRect requiresNucleusIcon = null!;
+    private Label processesLabel = null!;
+    private Label enzymesLabel = null!;
+    private Label massLabel = null!;
+    private Label sizeLabel = null!;
+    private Label osmoregulationCostLabel = null!;
+    private Label storageLabel = null!;
+    private TextureRect uniqueIcon = null!;
+    private Label upgradesLabel = null!;
+    private Label internalNameLabel = null!;
+
+    private Texture? modelTexture;
+    private Texture? modelLoadingTexture;
+#pragma warning restore CA2213
+
+    private ImageTask? modelImageTask;
+    private bool finishedLoadingModelImage;
+
+    /// <summary>
+    ///   The organelle this info box displays data for.
+    /// </summary>
+    public OrganelleDefinition? Organelle
+    {
+        get => organelle;
+        set
+        {
+            organelle = value;
+            UpdateValues();
+            UpdateModelImage();
+        }
+    }
+
+    public Texture? ModelTexture
+    {
+        get => modelTexture;
+        set
+        {
+            modelTexture = value;
+            UpdateModelImage();
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        nameLabel = GetNode<Label>(NamePath);
+        icon = GetNode<TextureRect>(IconPath);
+        model = GetNode<TextureRect>(ModelPath);
+        costLabel = GetNode<Label>(CostLabelPath);
+        requiresNucleusIcon = GetNode<TextureRect>(RequiresNucleusIconPath);
+        processesLabel = GetNode<Label>(ProcessesLabelPath);
+        enzymesLabel = GetNode<Label>(EnzymesLabelPath);
+        massLabel = GetNode<Label>(MassLabelPath);
+        sizeLabel = GetNode<Label>(SizeLabelPath);
+        osmoregulationCostLabel = GetNode<Label>(OsmoregulationCostLabelPath);
+        storageLabel = GetNode<Label>(StorageLabelPath);
+        uniqueIcon = GetNode<TextureRect>(UniqueIconPath);
+        upgradesLabel = GetNode<Label>(UpgradesLabelPath);
+        internalNameLabel = GetNode<Label>(InternalNameLabelPath);
+
+        modelLoadingTexture = GD.Load<Texture>("res://assets/textures/gui/bevel/IconGenerating.png");
+        UpdateModelImage();
```

Comment:
Sorry one more thing here. Shouldn't this also call `UpdateValues`? Because `Organelle` has that, so this still doesn't call all of the setup stuff at the end of _Ready.

Reply:
Added.